src/main/java/hometaskstream/Book.java

```diff
@@ -0,0 +1,142 @@
+        for (int i = 0; i < targetStringLength; i++) {
+            int randomLimitedInt = leftLimit + (int)
+                    (random.nextFloat() * (rightLimit - leftLimit + 1));
+            buffer.append((char) randomLimitedInt);
+        }
+        return buffer.toString();
+    }
+
+    public static int getRandomBookPrice() {
+        return (int) (Math.random() * 100);
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setPrice(int price) {
+        this.price = price;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public static List<Book> getRandomBooks(int size) {
+        List<Book> book = new ArrayList<>();
+        for (int i = 0; i < size; i++) {
+            book.add(new Book());
+        }
+        return book;
+    }
+
+    public static Collection<Book> printAllBooks(List<Book> bookList) {
+        return bookList;
+    }
+
+    public static List<String> printAllBooksName(List<Book> bookList) {
+        return bookList
+                .stream()
+                .map(Book::getName)
+                .collect(Collectors.toList());
+    }
+
+    public static Integer getTotalPriceOfAllBooks(List<Book> bookList) {
+        return bookList
+                .stream()
+                .map(Book::getPrice)
+                .reduce(0, Integer::sum);
+    }
+
+    public static Long getNumberOfBooksWithFiveLettersName(List<Book> booksList) {
+        return booksList
+                .stream()
+                .map(Book::getName)
+                .filter(bookName -> bookName.length() == 5).count();
+    }
+
+    public static String getBookNameOnALetter(List<Book> bookList) {
+        return bookList
+                .stream()
+                .map(Book::getName)
+                .filter(booksName -> booksName.startsWith("A") || booksName.startsWith("a"))
+                .findFirst()
+                .get();
+    }
+
+    public static String getBookWithHighestPrice(List<Book> bookList) {
+        return bookList
+                .stream()
+                .max(Comparator.comparing(Book::getPrice))
+                .map(Book::getName)
+                .get();
+    }
+
+    public static Collection<Book> getBookListThatPriceHigher30(List<Book> bookList) {
+        return bookList
+                .stream()
+                .filter(book -> book.getPrice() > 30)
+                .collect(Collectors.toList());
+    }
+
+    public static Collection<Book> getBookListInDescOrder(List<Book> bookList) {
+        return bookList
+                .stream()
+                .sorted(Comparator.comparing(Book::getName).reversed().thenComparingDouble(Book::getPrice))
+                .collect(Collectors.toList());
+    }
+
+    public static String getLibraryName(Book book1, Book book2) {
+        Map<Book, String> mapWithBook = new HashMap<>();
```

Comment:
it is the correct implementation, but please move this code to the test itself and delete this method. it is a test-case, actually, just the way we can verify that equals and hashCode are overridden correctly, we do not need this method in the Book class

Reply:
thanks, done

Reply:
now you can remove getLibraryName() method from the Book class and use your mapWithBook variable directly in the test:
<img width="1333" alt="image" src="https://user-images.githubusercontent.com/90613721/184716206-4a3b2902-2e61-4245-82ac-2834ef8b5e6e.png">

Reply:
thanks, done :)

Reply:
getLibraryName() method is still present in the Book class - please remove it

Reply:
sorry, thanks - method was removed